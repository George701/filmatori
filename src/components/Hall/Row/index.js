import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Place from '../Place';

const Row = ({ row: { _id, rowNumber, skip, places } }) => {

    const generateRow = places.map(place => (
        <Place key={place._id} place={place} row_id={_id} />
    ));

    return (
        <Fragment>
            <div className='row-main'>
                <span className='row-number'>{rowNumber}</span>
                <div className='row-places'>
                    {generateRow}
                </div>
                <span className='row-number'>{rowNumber}</span>
            </div>
            {skip && <div className='row-main'></div>}
        </Fragment>
    )
}

Row.propTypes = {
    row: PropTypes.object.isRequired,
}

export default Row;
