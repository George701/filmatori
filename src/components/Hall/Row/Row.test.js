import React from 'react';
import { shallow } from '../../../enzyme';
import Row from './index';

describe('Row', () => {
    const props = {
        row: {
            _id: 'RT01',
            rowNumber: 1,
            skip: true,
            places: [
                {
                    "_id": "PT09",
                },
                {
                    "_id": "PT10",
                },
            ]
        }
    };
    const row = shallow(<Row {...props} />);

    it('renders properly', () => {
        expect(row).toMatchSnapshot();
    });

    it('displays empty space for next row ', () => {
        expect(row.find('.row-main').at(1).exists()).toBe(true);
    });

    it('displays correct number of row number identifiers', () => {
        expect(row.find('.row-number').length).toEqual(2);
    });

    it('displays correct number of rows identifier', () => {
        expect(parseInt(row.find('.row-number').at(0).text(), 10)).toEqual(props.row.rowNumber);
        expect(parseInt(row.find('.row-number').at(1).text(), 10)).toEqual(props.row.rowNumber);
    });

    it('displays correct number of generated places', () => {
        expect(row.find('Connect(Place)').length).toEqual((props.row.places).length)
    });
});