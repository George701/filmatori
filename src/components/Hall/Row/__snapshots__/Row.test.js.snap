// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders properly 1`] = `ShallowWrapper {}`;
